### DISCUSSION TITLE
Laravel Cashier: trial info not in Stripe

### DISCUSSION THREAD

**Discussion by ChristianFenn** [laravel/framework#34448] [General] [answered]
When using Laravel Cashier I'm giving users a free "generic trial" in a registration endpoint. I create a user and set the `trial_ends_at` field to a value such as `2020-09-22 18:07:58`. This is basically the same example as shown in the [Cashier docs](https://laravel.com/docs/5.7/billing#without-credit-card-up-front).

Here is a simplified version of what I have:

```
$trialEndDate = now()->addDays(1); // Just set to 1 for wanting to test live webhooks quickly

$user = User::create([
	'name' => $request->get('name'),
	'email' => $request->get('email'),
         // ....
	'trial_ends_at' => $trialEndDate
]);
```

At this point, `$user->onTrial()` returns `true`. However, in the Stripe dashboard, I don't see any evidence that the user was created on a trial:

![Screen Shot 2020-09-21 at 11 42 34 AM](https://user-images.githubusercontent.com/53135800/93807508-9730a000-fbff-11ea-848c-8054647236e3.png)

It looks like Stripe isn't aware of users going on a generic trial like this, which ends up being an issue when you want to handle the trial ending via the `customer.subscription.updated` webhook. Is Stripe not being aware of generic trials the expected behavior? If not, what could be causing this?

Edit: I've also tried using `$user->createAsStripeCustomer();` after the user is initially saved, and with this change I can see the new customer in the Stripe dashboard, but still no evidence that they're on a trial. All actions post-trial work fine (purchasing subscriptions, updating subscriptions, cancelling, adding new payment methods, etc.). I'm more than happy to provide any additional information.

I'm using Laravel 7 and Cashier 12.2.

Thank you.

**Comment by GrahamCampbell** [comment, marked as answer]
> It looks like Stripe isn't aware of users going on a generic trial like this

Yes, that is the expected heaviour. Stripe only counts trials on a specific plan.

**Comment by driesvints** [comment]
Afaik Stripe doesn't has a concept of generic trialing.

**Comment by ChristianFenn** [comment]
Thanks for the replies! Cheers
